src/scripts/i18n/i18n_extraction.py

```
@@ -0,0 +1,364 @@
+    # Only write to failed_batches.jsonl if all retries failed
+    if batch is not None and batch_num is not None:
+        failed_batch = {
+            'batch_num': batch_num,
+            'files': {file_name: file_content for file_name, file_content in batch},
+            'error': str(last_exception)
+        }
+        with open('src/scripts/failed_batches.jsonl', 'a', encoding='utf-8') as fail_f:
+            fail_f.write(json.dumps(failed_batch, ensure_ascii=False) + '\n')
+    raise RuntimeError("Failed to call Gemini API after retries")
+
+def process_gemini_response(resp_content):
```

Ruff (0.11.9)

Missing return type annotation for public function `process_gemini_response`

(ANN201)